aws-datastore/src/main/java/com/amplifyframework/datastore/AWSDataStorePlugin.java

```diff
@@ -449,7 +450,10 @@ private Completable initializeOrchestrator() {
         if (api.getPlugins().isEmpty()) {
             return Completable.complete();
         } else {
-            return orchestrator.start();
+            // Let's prevent the orchestrator startup from possibly running in main.
+            return orchestrator.start()
+                .observeOn(Schedulers.io())
+                .subscribeOn(Schedulers.io());
```

Comment:
I saw this pattern in multiple place in Amplify Android's source code. Is there any reason why we want perform observeOn and subscribeOn at the same place with the same scheduler?

Should'nt 

```
orchestrator.start()
                  .subscribeOn(Schedulers.io());
```

Do the work? Unless you were using a Subject which may change the threading so you have to reset the scheduler in observeOn but i dont see any Subject in orchestrator

Reply:
The thinking behind that was to ensure it was not in main since we have the occasional blockingAwait/Get call throughout the process. Not sure if it makes a difference, but the subscriber of the `SubscriptionProcessor` is a `ReplaySubject`

Reply:
@rjuliano yeah then i dont think its needed to call observeOn again, it will just wrap the lower stream runnable into the same thread pool. 

 we only need to switch to same scheduler if we have below use case , which is to transform a callback style code into rxjava observable 

```
 Observable.<StorageItemChange<? extends Model>>create(emitter ->
                localStorageAdapter.observe(emitter::onNext, emitter::onError, emitter::onComplete)
            )
            .subscribeOn(Schedulers.single())
            .observeOn(Schedulers.single())

```
The above code makes sense cause the emitter will be emitting the onNext event in a different thread other than the one we specify in Schedulers.single(), so it makes sense to quickly switch back to it by calling single,

similar use case : https://github.com/aws-amplify/amplify-android/blob/main/aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/SubscriptionProcessor.java#L137

nit: the above link code i think it need to be changed as well as the onErrorResume() will be executed not in IO() thread scheduler

Reply:
That makes sense. That's the change you made in #612 right @richardmcclellan ?

Reply:
I think you meant @richardissuperman, not me :)

Reply:
@rjuliano yep, exactly, but that PR was messed up with my previous commit, will revise all two later tonight

Reply:
Cool...that's super helpful and thanks for updating the PR. I just merged it.